Server/src/models/DatasetModels/DatasetApprovalModel.ts

```diff
@@ -70,9 +71,12 @@ export class DatasetApprovalModel {
     }
 
     async updateDatasetComments(datasetId: number, datasetCommentsToAppend?: string) {
-        if (datasetCommentsToAppend !== null && datasetCommentsToAppend !== undefined) {
+        if (datasetCommentsToAppend !== undefined && datasetCommentsToAppend !== "undefined" && datasetCommentsToAppend !== null) {
             let oldComment = await this.selectDatasetCommentQuery(datasetId)
-            let newComment = oldComment.comments.concat(" " + datasetCommentsToAppend)
+            let newComment = datasetCommentsToAppend
+            if (oldComment.comments !== null) {

```

Comment:
same here if (oldComment.comments) <- should do the job

Reply:
This might work, will check.